app/src/utils/services/DataService.ts

```diff
@@ -1,55 +1,40 @@
 import axios from 'axios';
-import QueryBuilder, { QueryObject } from 'es-query-builder';
+import QueryBuilder from 'es-query-builder';
 import Location from '@/models/location';
+import Post from '@/models/post';
+
+export interface PaginatedResponse<T> {
+    data: T[];
+    meta: {
+        total: number;
+        size: number;
+    };
+}
+
+export interface SearchParameters {
+    searchValues: string[];
+    location: Location|undefined;
+    radius: string;
+    from: number;
+    size: number;
+}
 
 class DataService {
     private baseUrl = 'https://cai-einander-helfen.fbi.h-da.de/api/_search/';
 
-    public findById<T>(id: string): Promise<T> {
-        const queryData = new QueryBuilder()
-            .mustTerm('_id', id);
-        return this.performQuery<T>(queryData);
-    }
+    public findBySelection(params: SearchParameters): Promise<PaginatedResponse<Post>> {
+        const {
+            searchValues,
+            location,
+            radius,
+            size,
+            from
+        } = params;
 
-    public findByCategory<T>(category: string): Promise<T> {
-        const queryData = new QueryBuilder()
-            .mustMatch('categories', `${category}`);
-        return this.performQuery<T>(queryData);
-    }
-
-    public findByTitle<T>(title: string): Promise<T> {
-        const queryData = new QueryBuilder()
-            // TODO Matching both if string contains the keyword für
-            .mustWildcard('title', `${title}`);
-        return this.performQuery<T>(queryData);
-    }
-
-    public findByWildcard<T>(object: string): Promise<T> {
-        const queryData = new QueryBuilder()
-            // TODO Matching both if string contains the keyword für
-            .size(15)
-            .shouldWildcard('title', `${object}`)
-            .shouldMatch('categories', `${object}`)
-            .shouldWildcard('organization', `${object.toLowerCase()}`);
-        return this.performQuery<T>(queryData);
-    }
-
-    public findAll<T>(): Promise<T> {
-        const queryData = new QueryBuilder();
-        return this.performQuery<T>(queryData);
-    }
-
-    public findBySelection({searchValues,
-                               location,
-                               radius
-                            }: {searchValues: string[],
-                                location: Location|undefined,
-                                radius: string
-                            }): Promise<any> {
         const query = new QueryBuilder();
         query.mustShouldMatch(searchValues.map((value) => [{key: 'title', value}, {key: 'categories', value}]).flat());
-
-        query.size(100);
+        query.from(from);
```

Comment:
The compiler throws an error on this line.

Reply:
Please update your installed dependencies by running `npm i`

Reply:
![Bildschirmfoto_2020-06-25_20-22-30](https://user-images.githubusercontent.com/64203299/85777305-cbd8ac80-b721-11ea-8121-90cd93563318.png)
I still get this error.

I am using an arch linux.

Reply:
Could you please delete your node_modules dir `rm -rf node_modules` and recreate it via `npm i` this seems to be a npm caching issue. We had similiar problems some days ago on the openArtBrowser.

Our CI/CD pipeline complete´s this build without problem: https://github.com/hochschule-darmstadt/einander-helfen/pull/91/checks?check_run_id=807562036

Reply:
The clean install fixed the issue. Thanks.